src/server/generic_family.cc

```diff
@@ -587,7 +587,17 @@ bool ScanCb(const OpArgs& op_args, PrimeIterator prime_it, const ScanOpts& opts,
   }
 
   bool matches = !opts.type_filter || it->second.ObjType() == opts.type_filter;
-
+  if (opts.mask.has_value()) {
+    if (opts.mask == ScanOpts::Mask::Volatile) {
+      matches = it->second.HasExpire();
+    } else if (opts.mask == ScanOpts::Mask::Permanent) {
+      matches = !it->second.HasExpire();
+    } else if (opts.mask == ScanOpts::Mask::Accessed) {
+      matches = it->second.WasTouched();
```

Comment:
if you check db_slice.cc:624 you will see that Touched attribute refers to the key, i.e. it->first